### PR TITLE
reinstall changes to `rose-suite.conf` [tests]

### PR DESCRIPTION
**Pull request by wxtim**
These changes contain a test for the issue described in https://github.com/cylc/cylc-flow/pull/5125

## Requirements check-list

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to `setup.cfg`.
- [x] Is a test
- [x] No change log entry required - is a test
- [x] No documentation update required.

### CODE REVIEW

**Review by oliver-sanders — APPROVED** [review]
Confirmed test fails when run with `cylc-flow:master` but passes on your branch.

**Review by datamel — APPROVED** [review]

**Comment by oliver-sanders** [comment]
(no need to merge into 1.1.1, get tests to pass in CI before merging)

**Comment by oliver-sanders** [comment]
Test now appears to be failing on https://github.com/cylc/cylc-flow/pull/5125

Possibly something to do with the branch change from master to 8.0.x?

**Review by oliver-sanders — APPROVED** [review]

**Review by oliver-sanders — CHANGES_REQUESTED** [review]
The reason this test is failing is because change you require has not yet been merged to master - 
https://github.com/cylc/cylc-flow/pull/5178

The "fix test" commit defeats the purpose of the test?

File: tests/functional/test_reinstall.py [hunk 1]
```diff
@@ -317,5 +317,5 @@ def test_reinstall_workflow(tmp_path):
     stdout = reinstall_workflow(
         source_dir, "flow-name", run_dir, dry_run=True)
 
-    expect = sorted(['send rose-suite.conf', 'send flow.cylc'])
+    expect = sorted(['send flow.cylc'])
```
[the remark this answers is not in the document]
Comment (wxtim) [reply tests/functional/test_reinstall.py:320]:
```suggestion
    expect = sorted(['send rose-suite.conf', 'send flow.cylc'])
```

**Comment by wxtim** [comment]
> The reason this test is failing is because change you require has not yet been merged to master - [cylc/cylc-flow#5178](https://github.com/cylc/cylc-flow/pull/5178)
> 
> The "fix test" commit defeats the purpose of the test?

Oh dear, I'm not looking very clever with all this multiple branch stuff. 🙄

**Review by oliver-sanders — APPROVED** [review]
Confirmed this passes after the cylc-flow fix but fails before it.

**Pull request merged**